shared-data/js/fixtures.ts

```diff
@@ -164,6 +256,142 @@ export function getAddressableAreaFromSlotId(
+  }
+  return SINGLE_CENTER_SLOT_FIXTURE
+}
+
+export const getReplacementFixtureForFakeFixture = (
+  cutoutFixtureId: CutoutFixtureIdsWithFakes
+): CutoutFixtureId => {
+  if (cutoutFixtureId === FAKE_STAGING_AREA_RIGHT_SLOT) {
+    return SINGLE_RIGHT_SLOT_FIXTURE
+  }
+  return cutoutFixtureId
+}
+
+export const replaceStagingFixtureAndTransformCutoutFixturesToAA = (
+  cutoutFixtures: CutoutConfig[],
+  deckDefinition: DeckDefinition
+): CutoutConfigMap[] => {
+  return cutoutFixtures.reduce<CutoutConfigMap[]>((acc, obj) => {
+    const cutoutFixtureReplacment = getCutoutFixtureReplacementIfNeeded(
+      obj.cutoutFixtureId,
+      deckDefinition
+    )
+
+    const aaPerCutoutFixture = getAAFromCutoutFixtureId(
+      obj.cutoutId,
+      cutoutFixtureReplacment,
+      deckDefinition
+    )
+    aaPerCutoutFixture?.forEach(item => {
+      acc.push({
+        ...obj,
+        addressableAreaId: item,
+        cutoutFixtureId: cutoutFixtureReplacment,
+      })
+    })
+    return acc
+  }, [])
+}
+
+export const filterAAByAreaType = (
+  cutoutFixtures: CutoutConfigMap[],
+  deckDef: DeckDefinition,
+  areaType: AreaTypeWithFakes
+): CutoutConfigMap[] => {
+  const deckDefWithFakeLocations = getDeckDefAAWithFakeAA(deckDef)
+  return cutoutFixtures.filter(({ addressableAreaId }) => {
+    return deckDefWithFakeLocations.locations.addressableAreas.find(
+      aa => aa.id === addressableAreaId && aa.areaType === areaType
+    )
+  })
+}
+
+const getDeckDefAAWithFakeAA = (
+  deckDefinition: DeckDefinition
+): DeckDefinitionWithFakes => {
+  const locationsWithFakeAA: AddressableAreaWithFakes[] = deckDefinition.locations.addressableAreas.concat(
+    FAKE_FIXTURES_AND_AA.locations.addressableAreas as any
+  )
+  return {
+    ...deckDefinition,
+    locations: {
+      ...deckDefinition.locations,
+      addressableAreas: locationsWithFakeAA,
+    },
+  }
+}
+
+export const getAALocationForCutoutAndFixtureId = (
+  addressableArea: AddressableAreaNamesWithFakes,
+  deckDefinition: DeckDefinition
+): CoordinateTuple => {
+  const deckDefWithFakeLocations = getDeckDefAAWithFakeAA(deckDefinition)
+  const addressableAreaItem = deckDefWithFakeLocations.locations.addressableAreas.find(
+    (aaItem: AddressableAreaWithFakes) => aaItem.id === addressableArea
+  )
+  if (addressableAreaItem == null) {
+    console.error(`Addressable area ${addressableArea} location was not found.`)
+  }
+  return addressableAreaItem?.offsetFromCutoutFixture ?? [0, 0, 0]
+}
+
+export const getAAFromCutoutFixtureId = (
+  inputCutoutId: CutoutId,
+  cutoutFixtureId: CutoutFixtureIdsWithFakes,
+  deckDefinition: DeckDefinition
+): AddressableAreaNamesWithFakes[] | null => {
+  /**
+   * Given a cutoutId and a cutoutFixtureId, returns a list of AA, or null if there is none
+   */
+  const cutoutFixturesWithFakeFixtures: CutoutFixtureWithFakes[] = deckDefinition.cutoutFixtures.concat(
+    FAKE_FIXTURES_AND_AA.cutoutFixtures as any
+  )
```

Comment:
```suggestion
  ]
```
switching this from a call to `concat` to a spread should let typescript infer it properly and may let you avoid the type assignment to the name binding in the first line